### DISCUSSION TITLE
Order of DS18b20 sensors.

### DISCUSSION THREAD

**Discussion by Papablaubaer** [arendst/Tasmota#15923] [Q&A] [answered]
Are several DS18b20 always sorted in the same way, or would it be conceivable that a sensor is once DS18b20-1, after the next restart DS18b20-3

**Comment by barbudor** [comment, marked as answer]
Order is totally random
You can compile your own binary with `DS18x20_USE_ID_AS_NAME` to get an fixed ID per sensor
This is not available by default in prebuilt binary

**Reply by sfromis** [reply]
If you instead rely on the `"Id"` part of the payload, you get a stable identification without a custom build.

**Comment by Papablaubaer** [comment]
My idea was to write a rule that reacts to the values ​​of a certain sensor. In this case, it seems difficult, if not impossible, to me. In my test, the sensors are always sorted by ID in ascending order, but that's probably coincidental.

**Reply by sfromis** [reply]
It may not be strictly coincidental (random), but there is no "promise" of a specific order. And if one sensor for whatever reasons fails to register at restart time, following sensors will absolutely get a different number. To me, the number suffix for names is only for the purpose of having unique names, not that you always knows what `#2` is, no matter what.

**Comment by Papablaubaer** [comment]
That's how I understood it. My idea was, if my number of sensors is available, sensor1 is always sensor1. If the number is not correct, the assignment will also change, but that is probably not completely certain either. Thank you for your help, I think it's great what you're doing here.
